### DISCUSSION TITLE
Train the teacher and student end-to-end in an iterative fashion

### DISCUSSION THREAD

**Discussion by Houssem171** [SforAiDl/KD_Lib#134] [Q&A] [answered]
is there a way to do an iterative training where the teacher can learn from the student's feedback?

**Comment by NeelayS** [comment, marked as answer]
Hi @Houssem171, we currently do not have such a method implemented in the library. Do you any specific paper/method in mind?

**Reply by Houssem171** [reply]
the method is mentioned in this paper  "Emerging Properties in Self-Supervised Vision Transformers."
link : https://openaccess.thecvf.com/content/ICCV2021/papers/Caron_Emerging_Properties_in_Self-Supervised_Vision_Transformers_ICCV_2021_paper.pdf

**Reply by NeelayS** [reply]
Unfortunately, we don't have an implementation of this paper in the library and neither is it straightforward for a user to extend the library's methods to quickly implement it. 
If you wish, you can create an issue for this and we'll try to get around to implementing it.
